src/status_im/chat/models/input.cljs

```diff
@@ -134,6 +134,33 @@
              (update-in [:chat/inputs current-chat-id :metadata]
                         dissoc :sending-image))}))
 
+(fx/defn soft-delete-message-success
+  {:events [::soft-delete-message-success]}
+  [{:keys [db]} chat-id message-id]
+  (let [new-last-message (-> db
+                             :messages
+                             (get chat-id)
+                             vals
+                             (->> (sort-by :whisper-timestamp)
+                                  (filter #(and (not (:hidden-in-ui? %))
```

Comment:
I think  you might not need a `hidden-in-ui?` field, we have already `deleted` I believe, and there's a flag we can expose as well to the JSON `hide`.
Also I think the logic of replacing the last message should go in status-go, we should read the response from status go (MessengerResponse) and just process that as we would with any signal.

Reply:
Got rid of `hidden in ui` will see how to move login to status-go.